src/components/widgets/DateRange.vue

```diff
@@ -0,0 +1,79 @@
+<script setup lang="ts">
+import { ref, watch } from 'vue';
+import { DateRange } from '@/constants/range';
+import EditableDate from '@/components/widgets/EditableDate.vue';
+
+interface DateRangeModel {
+  type: string | undefined;
+  start: string | undefined;
+  end: string | undefined;
+}
```

Comment:
```suggestion
type DateRangeModel {
  type?: DateRange;
  start?: string;
  end?: string;
}
```

It may make sense to make DateRangeTypeEnum